src/gluonts/model/seasonal_naive/_predictor.py

```diff
@@ -54,18 +59,25 @@ class SeasonalNaivePredictor(RepresentablePredictor):
     def __init__(
         self,
         prediction_length: int,
-        season_length: int,
+        season_length: Union[int, Callable],
         imputation_method: MissingValueImputation = LastValueImputation(),
     ) -> None:
         super().__init__(prediction_length=prediction_length)
 
-        assert season_length > 0, "The value of `season_length` should be > 0"
+        assert (
+            not isinstance(season_length, int) or season_length > 0
+        ), "The value of `season_length` should be > 0"
 
         self.prediction_length = prediction_length
         self.season_length = season_length
         self.imputation_method = imputation_method
 
     def predict_item(self, item: DataEntry) -> Forecast:
+        if isinstance(self.season_length, int):
+            season_length = self.season_length
+        else:
+            season_length = self.season_length(item["start"].freq)
```

Comment:
This requires that there is a `freq` for the time series, which might not be the case in the future?

Reply:
I guess this mechanism is really only useful if you have frequency information attached to the series. If that's not the case, there's no point in using this option, and the model should be configured with an `int` value.

Also: I believe the assumption that `item["start"]` is a `pd.Period` (hence has frequency information) is already baked into this method since it uses [`forecast_start`](https://github.com/awslabs/gluonts/blob/a25ca9dc61cb22f426dbc6004e19394b8130766e/src/gluonts/dataset/util.py#L21)?

Reply:
At the moment we require that all time series have frequency information attached -- but that's an arbitrary requirement.